src/main/java/org/sunbird/extension/util/TransformJsonUtil.java

```diff
@@ -0,0 +1,322 @@
+
+    Map<String, Object> outputMap = new HashMap<String, Object>();
+    config = fieldsConfig;
+    Set<String> userInputSet = userInputMap.keySet();
+    Iterator<String> userInputIterator = userInputSet.iterator();
+    String prefix = rootConfig + TransformationConstants.DOT;
+    while (userInputIterator.hasNext()) {
+      String key = userInputIterator.next();
+      try {
+        Map<String, Object> fieldMap = (Map<String, Object>) config.getAnyRef(prefix + key);
+        if (!fieldMap.containsKey(TransformationConstants.FROMFIELDNAME)) {
+          fieldMap.put(TransformationConstants.FROMFIELDNAME, key);
+        }
+        transformField(fieldMap, userInputMap, outputMap);
+      } catch (Exception e) {
+        // handle exception
+      }
+    }
+
+    return outputMap;
+  }
+
+  private void transformField(
+      Map<String, Object> fieldMap,
+      Map<String, Object> userInputMap,
+      Map<String, Object> outputMap) {
+
+    Object fieldValue = null;
+    String fromField = null;
+    List<String> fromFields = null;
+    String toField = null;
+    String fromType = null;
+    String toType = null;
+
+    toField = (String) fieldMap.get(TransformationConstants.TOFIELDNAME);
+    fromType = (String) fieldMap.get(TransformationConstants.FROMTYPE);
+    toType = (String) fieldMap.get(TransformationConstants.TOTYPE);
+    if (fieldMap.get(TransformationConstants.FROMFIELDNAME) instanceof String) {
+      fromField = (String) fieldMap.get(TransformationConstants.FROMFIELDNAME);
+      fieldValue = getValueFromIncomingMap(fromField, userInputMap);
+    } else if (fieldMap.get(TransformationConstants.FROMFIELDNAME) instanceof List) {
+      fromFields = (List<String>) fieldMap.get(TransformationConstants.FROMFIELDNAME);
+      fieldValue = getValueFromIncomingMap(fromFields, userInputMap);
+    }
+
+    if (!fromType.equalsIgnoreCase(toType) || fieldMap.containsKey(TransformationConstants.ENUM)) {
+      Object enumName = fieldMap.get(TransformationConstants.ENUM);
+      Map<String, Object> enumValues = null;
+      if (enumName instanceof String) {
+        enumValues =
+            (Map<String, Object>)
+                config.getAnyRef(
+                    TransformationConstants.ENUMS + TransformationConstants.DOT + enumName);
+      } else if (enumName instanceof Map) {
+        enumValues = (Map<String, Object>) enumName;
+      }
+      fieldValue = transformValue(fromType, toType, fieldValue, enumValues, fieldMap);
+    }
+
+    putValueIntoOutgoingMap(toField, toType, fieldValue, outputMap);
+  }
+
+  private Object transformValue(
+      String fromType,
+      String toType,
+      Object fieldValue,
+      Map<String, Object> enumValues,
+      Map<String, Object> fieldMap) {
+
+    if (!fromType.contains(TransformationConstants.LIST)
+        && !toType.contains(TransformationConstants.LIST)) {
+      fieldValue = convertSimpleTypes(fromType, toType, fieldValue, enumValues, fieldMap);
```

Comment:
Rename as follows.

convertSimpleTypes to getTransformedFieldValueSimpleType
convertListTypes to getTransformedFieldValueListType
convertCustomListTypes to getTransformedFieldValueCustomListType
convertListtoSimpleType to getTransformedFieldValueListToSimpleType
convertSimpletoListType to getTransformedFieldValueSimpleToListType